examples/runners/wgpu/build.rs

```diff
@@ -1,15 +1,24 @@
-use spirv_builder::SpirvBuilder;
 use std::error::Error;
 
-fn build_shader(path_to_create: &str) -> Result<(), Box<dyn Error>> {
-    SpirvBuilder::new(path_to_create, "spirv-unknown-vulkan1.0").build()?;
-    Ok(())
-}
-
 fn main() -> Result<(), Box<dyn Error>> {
-    build_shader("../../shaders/sky-shader")?;
-    build_shader("../../shaders/simplest-shader")?;
-    build_shader("../../shaders/compute-shader")?;
-    build_shader("../../shaders/mouse-shader")?;
+    let target_os = std::env::var("CARGO_CFG_TARGET_OS")?;
+    let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH")?;
+    // Always build on CI to make sure the shaders can still build
+    let is_on_ci = std::env::var("CI");
+    if target_os != "android" && target_arch != "wasm32" && is_on_ci.is_err() {
+        return Ok(());
+    }
+    let status = std::process::Command::new("cargo")
+        .args(["run", "--release", "-p", "example-runner-wgpu-builder"])
+        .stderr(std::process::Stdio::inherit())
+        .stdout(std::process::Stdio::inherit())
+        .status()?;
```

Comment:
Doesn't this deadlock Cargo without passing a different `--target-dir`? Have you tried building `example-runner-wgpu` with `--release` and `CI` being set? It should get stuck AFAICT (given e.g. `touch examples/runners/wgpu/builder/src/main.rs` to force a rebuild of it).